egs/thchs30/s5/local/dae/run_dae.sh

```diff
@@ -0,0 +1,149 @@
+#For dev, generate noisy data with SNR mean=0, variance=0, with three niose types mixed together
+#For test, use the standard test data which were generated by SNR mean=0, variance=0.
+
+if [ $stage =  0 ]; then
+   #generat noise.scp
+   mkdir -p data/dae/noise && \
+   awk '{print $1 " '$thchs'/resource/noise/"$2}' $thchs/resource/noise/noise.scp >  data/dae/noise/noise.scp || exit 1
+
+   echo "DAE: generate training data..."
+   noise_scp=data/dae/noise/noise.scp
+   noise_prior="0.0,10.0,10.0,10.0" #define noise type to sample. [S_clean, S_white, S_car, S_cafe]
+   noise_level=0 #0db condition
+   sigma0=10 #some random in SNR
+   seed=32
+   verbose=0
+   wavdir=wav/dae/train 
+   rm -rf data/dae/train && mkdir -p data/dae/train || exit 1
+   cp data/fbank/train/{spk2utt,utt2spk,text} data/dae/train || exit 1
+   mkdir -p $wavdir && awk '{print $1 " '$wavdir'/"$1".wav"}' data/fbank/train/wav.scp > data/dae/train/wav.scp || exit 1
+
+   mkdir -p exp/dae/gendata 
+   split_scps=""
+   for n in $(seq $nj); do
+      split_scps="$split_scps exp/dae/gendata/train_split_${n}.scp"
+   done
+   utils/split_scp.pl data/fbank/train/wav.scp  $split_scps || exit 1
+   $train_cmd JOB=1:$nj exp/dae/gendata/add_noise_train.JOB.log \
+     local/dae/add-noise-mod.py --noise-level $noise_level \
+       --sigma0 $sigma0 --seed $seed --verbose $verbose \
+       --noise-prior $noise_prior --noise-src $noise_scp \
+       --wav-src exp/dae/gendata/train_split_JOB.scp --wavdir $wavdir \
+       || exit 1
+
+   steps/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
+     data/dae/train exp/dae/gendata fbank/dae/train || exit 1
+   steps/compute_cmvn_stats.sh data/dae/train exp/dae/cmvn \
+     fbank/dae/train/_cmvn || exit 1
+
+   #genreate dev data. Just the 0db condition is produced.  Multiple noise types mixed together.
+   echo "DAE: generating dev data..."
+   wavdir=wav/dae/dev/0db
+   sigma0=0 #no random in SNR
+   rm -rf data/dae/dev/0db && mkdir -p data/dae/dev/0db && \
+   cp -L data/fbank/dev/{spk2utt,utt2spk,text} data/dae/dev/0db || exit 1
+   mkdir -p $wavdir && awk '{print $1 " '$wavdir'/"$1".wav"}' data/fbank/dev/wav.scp > data/dae/dev/0db/wav.scp || exit 1
+
+   split_scps=""
+   for n in $(seq $nj); do
+      split_scps="$split_scps exp/dae/gendata/dev_split_${n}.scp"
+   done
+   utils/split_scp.pl data/fbank/dev/wav.scp  $split_scps || exit 1
+
+   $train_cmd JOB=1:$nj exp/dae/gendata/add_noise_dev.JOB.log \
+     local/dae/add-noise-mod.py --noise-level $noise_level \
+       --sigma0 $sigma0 --seed $seed --verbose $verbose \
+       --noise-prior $noise_prior --noise-src $noise_scp \
+       --wav-src exp/dae/gendata/dev_split_JOB.scp --wavdir $wavdir \
+       || exit 1
+   steps/make_fbank.sh --nj $nj --cmd "$train_cmd"  \
+     data/dae/dev/0db exp/dae/gendata fbank/dae/dev/0db || exit 1
+   steps/compute_cmvn_stats.sh data/dae/dev/0db exp/dae/cmvn \
+     fbank/dae/dev/0db/_cmvn || exit 1
```

Comment:
I just noticed this-- please remove all directory extensions of the type
x/y/_cmvn
(This is unnecessary and odd-looking).   Leave them as just x/y/.